elastic/datadog_checks/elastic/elastic.py

```diff
@@ -439,6 +461,123 @@ def _process_cat_allocation_data(self, admin_forwarder, version, base_tags):
+            dynamic_tag_value = value
+
+            # Traverse down the tree to find the tag value
+            for key in re.split(REGEX, dynamic_tag_path):
+                if dynamic_tag_value is not None:
+                    dynamic_tag_value = dynamic_tag_value.get(key.replace('\\', ''))
+                else:
+                    self.log.debug("Dynamic tag not found: %s -> %s", path, dynamic_tag_name)
+                    break
+
+            # If tag is there, then add it to list of tags to submit
+            if dynamic_tag_value is not None:
+                dynamic_tag = '{}:{}'.format(dynamic_tag_name, dynamic_tag_value)
+                tags_to_submit.append(dynamic_tag)
+            else:
+                self.log.debug("Dynamic tag is null: %s -> %s", path, dynamic_tag_name)
+
+        # Now do the same for the actual metric
+        branch_value = value
+        for key in re.split(REGEX, value_path):
+            if branch_value is not None:
+                branch_value = branch_value.get(key.replace('\\', ''))
+            else:
+                break
+
+        if branch_value is not None:
+            if xtype == "gauge":
+                self.gauge(metric_name, branch_value, tags=tags_to_submit, hostname=hostname)
+            elif xtype == "monotonic_count":
+                self.monotonic_count(metric_name, branch_value, tags=tags_to_submit, hostname=hostname)
+            else:
+                self.rate(metric_name, branch_value, tags=tags_to_submit, hostname=hostname)
+        else:
+            self.log.debug("Metric not found: %s -> %s", path, metric_name)
+
+    def _run_custom_queries(self, admin_forwarder, base_tags):
+        self.log.debug("Running custom queries")
+        custom_queries = self._config.custom_queries
+
+        for endpoints in custom_queries:
+            columns = endpoints.get('columns', [])
+            data_path = endpoints.get('data_path')
+            endpoint = endpoints.get('endpoint')
+            static_tags = endpoints.get('tags', [])
+            endpoint = self._join_url(endpoint, admin_forwarder)
+            payload = endpoints.get('payload', {})
+
+            data = self._get_data(endpoint, data=payload)
+
+            # If there are tags, add the tag path to list of paths to evaluate while processing metric
+            dynamic_tags = get_dynamic_tags(columns)
+            tags = base_tags + static_tags
+
+            for column in columns:
+                metric_type = column.get('type', 'gauge')
+
+                # Skip tags since already processed
+                if metric_type == 'tag':
+                    continue
+                name = column.get('name')
+                value_path = column.get('value_path')
+                if name and value_path:
+                    value = data
+
+                    # Traverse the nested dictionaries to the data_path
+                    for key in re.split(REGEX, data_path):
+                        if value is not None:
```

Comment:
in what cases is `value` `None`?

Reply:
I kept the logic the same from the other times where the data is parsed from the JSON response. I can move this for-loop into its own function since it's the same as the other ones though.